# - 준비물 : 전처리가 끝난 train, test data

## 1. 데이터 불러오기

In [56]:
df = pd.read_csv('data/train_ratio_0728_dash.csv') # train data 넣기

In [57]:
df.head()

단지코드    총세대수    아파트비율     상가비율   공가수  지하철역  버스정류장  단지내주차면수   등록차량수  \
0  C1000   566.0  1.00000  0.00000  10.0   0.0    1.0    438.0   481.0   
1  C1004   521.0  0.96737  0.03263   3.0   0.0    2.0    153.0    93.0   
2  C1005  1144.0  1.00000  0.00000  16.0   0.0    7.0    950.0   376.0   
3  C1013  1308.0  1.00000  0.00000  16.0   0.0    6.0   1119.0  1665.0   
4  C1014   996.0  1.00000  0.00000   5.0   0.0    2.0    823.0   708.0   

   전용면적_15_비율  전용면적_20_비율  전용면적_25_비율  전용면적_30_비율  전용면적_35_비율  전용면적_40_비율  \
0    0.000000    0.000000    0.000000    0.000000    0.740283         0.0   
1    0.005758    0.013436    0.001919    0.003839    0.971209         0.0   
2    0.000000    0.000000    0.000000    0.000000    0.000000         0.0   
3    0.000000    0.000000    0.000000    0.000000    0.222477         0.0   
4    0.000000    0.000000    0.000000    0.518072    0.000000         0.0   

   전용면적_45_비율  전용면적_50_비율  전용면적_55_비율  전용면적_60_비율  전용면적_65_비율  전용면적_70_비율  \
0    0.127208    0.132509    0.000000         0.0         0.0         0.0   
1    0.000000    0.001919    0.000000         0.0         0.0         0.0   
2    0.000000    0.790210    0.209790         0.0         0.0         0.0   
3    0.578746    0.198777    0.000000         0.0         0.0         0.0   
4    0.281124    0.116466    0.084337         0.0         0.0         0.0   

   전용면적_75_비율  전용면적_80_비율    초장기비율  장기비율     단기비율  ABEGH비율  A비율  ACEFI비율  \
0         0.0    0.000000  0.00000   1.0  0.00000      1.0  0.0  0.00000   
1         0.0    0.001919  0.96737   0.0  0.03263      0.0  0.0  0.96737   
2         0.0    0.000000  0.00000   1.0  0.00000      1.0  0.0  0.00000   
3         0.0    0.000000  0.00000   1.0  0.00000      1.0  0.0  0.00000   
4         0.0    0.000000  0.00000   1.0  0.00000      1.0  0.0  0.00000   

       D비율  JKLMNO비율  10대미만(여자)  10대미만(남자)   10대(여자)   10대(남자)   20대(여자)  \
0  0.00000       0.0   0.034678   0.035339  0.059808  0.061570  0.060824   
1  0.03263       0.0   0.031369   0.031711  0.059077  0.062422  0.067975   
2  0.00000       0.0   0.031369   0.031711  0.059077  0.062422  0.067975   
3  0.00000       0.0   0.031994   0.034566  0.049851  0.050254  0.065084   
4  0.00000       0.0   0.028089   0.029065  0.059685  0.060080  0.066262   

    20대(남자)   30대(여자)   30대(남자)   40대(여자)   40대(남자)   50대(여자)   50대(남자)  \
0  0.064937  0.061069  0.056625  0.082318  0.072648  0.082747  0.074276   
1  0.072622  0.065095  0.067303  0.078860  0.073418  0.079299  0.069167   
2  0.072622  0.065095  0.067303  0.078860  0.073418  0.079299  0.069167   
3  0.066875  0.066888  0.064416  0.080028  0.079183  0.079902  0.067443   
4  0.070322  0.052027  0.046596  0.077005  0.066645  0.081408  0.074382   

    60대(여자)   60대(남자)   70대(여자)   70대(남자)   80대(여자)   80대(남자)   90대(여자)  \
0  0.075390  0.062427  0.041814  0.027566  0.027762  0.011212  0.005386   
1  0.070278  0.057692  0.042296  0.028682  0.024514  0.010900  0.005429   
2  0.070278  0.057692  0.042296  0.028682  0.024514  0.010900  0.005429   
3  0.075930  0.059586  0.048552  0.031754  0.029749  0.010341  0.006343   
4  0.076636  0.068042  0.051025  0.035748  0.035049  0.012641  0.007223   

    90대(남자)  100대(여자)  100대(남자)    1인당자동차    연면적  평균분양가격      총면적  \
0  0.001310  0.000257  0.000037  0.523499  11343    2305  21655.0   
1  0.001549  0.000219  0.000123  0.526493   7879    2640  18110.0   
2  0.001549  0.000219  0.000123  0.526493   7879    2640  58400.0   
3  0.000895  0.000353  0.000013  0.464329  13393    3834  57250.0   
4  0.001898  0.000158  0.000013  0.510777  10146    2468  38500.0   

         규모별가격  강원도  경기도  경상남도  경상북도  광주광역시  대구광역시  대전광역시  부산광역시  서울특별시  \
0   49914775.0    0    0     0     0      0      0      0      0      0   
1   47810400.0    0    0     0     0      0      0      0      0      0   
2  154176000.0    0    0     0     0      0      0      0      0      0   
3  219496500.0    0    0     0     0      1      0      0      0      0   
4   95018000.0    0    0   

In [58]:
print("pandas version: ", pd.__version__)
pd.set_option('display.max_row', 100)
pd.set_option('display.max_columns', 85)

pandas version:  1.3.0


## 2. 필요 모듈 불러오기

In [73]:
from sklearn.linear_model import LinearRegression   ## 회귀 분석
from sklearn.svm import SVC                         ## svm 분류모델
from sklearn.tree import DecisionTreeClassifier     ## Tree 분류모델
from sklearn.ensemble import RandomForestClassifier ## RandomForet
from sklearn.linear_model import LogisticRegression ## Logistic Regression
from sklearn.ensemble import RandomForestRegressor  ## RandomForet 
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression, Ridge, Lasso, ElasticNet
import statsmodels.api as sm
from sklearn.metrics import mean_absolute_error
from xgboost import XGBRegressor
from sklearn.ensemble import GradientBoostingRegressor
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import KFold
from catboost import CatBoostRegressor
from sklearn.ensemble import ExtraTreesRegressor
from sklearn.ensemble import AdaBoostRegressor 
from sklearn.svm import SVR
from lightgbm import LGBMRegressor
from sklearn.ensemble import StackingRegressor
import pickle
from sklearn.ensemble import VotingRegressor

## 3. X, y

In [60]:
# 데이콘용 모델 생성시 포함
# df = df.drop('전용면적_65_세대수', axis=1)
# df = df.drop('서울특별시', axis=1)

In [61]:
x = df.columns.difference(['등록차량수'], sort=False).difference(['단지코드'], sort=False)
X= df[x]
y = pd.DataFrame(df['등록차량수'])

In [62]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

## 4. 모델링

In [63]:
cols1 = ['Linear', 'ExtraTrees', 'Gradient', 'RandomForest', 'ElasticNet', 'Ridge', 'Lasso', 'XGB', 'Catboost', 'Voting1', 'Voting2', 'Voting3', 'Voting4', 'Voting5']

In [64]:
ls1=[]
ls2=[]
lr=LinearRegression()
dt = ExtraTreesRegressor(random_state=1)
gbr = GradientBoostingRegressor(random_state=1)
rfr = RandomForestRegressor(random_state=1)
ela = ElasticNet(random_state=1)
rg = Ridge(random_state=1)
las = Lasso(random_state=1)
xg = XGBRegressor(random_state=1)
cb = CatBoostRegressor(depth = 6, iterations=100, learning_rate=0.1)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.20, random_state = 0)
lr.fit(X_train, y_train)
dt.fit(X_train, y_train)
gbr.fit(X_train, y_train)
rfr.fit(X_train,y_train)
ela.fit(X_train, y_train)
rg.fit(X_train,y_train)
las.fit(X_train,y_train)
xg.fit(X_train,y_train)
cb.fit(X_train, y_train)
ereg1 = VotingRegressor([('xg', xg), ('ela', ela), ('las', las)])
ereg1.fit(X_train, y_train)
ereg2 = VotingRegressor([('rfr', rfr), ('ela', ela), ('las', las)])
ereg2.fit(X_train, y_train)
ereg3 = VotingRegressor([('cb', cb), ('ela', ela), ('las', las)])
ereg3.fit(X_train, y_train)
ereg4 = VotingRegressor([('dt', dt), ('ela', ela), ('las', las)])
ereg4.fit(X_train, y_train)
ereg5 = VotingRegressor([('gbr', gbr), ('ela', ela), ('las', las)])
ereg5.fit(X_train, y_train)
    
ls1.append(round(mean_absolute_error(y_test, lr.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, dt.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, gbr.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, rfr.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, ela.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, rg.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, las.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, xg.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, cb.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, ereg1.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, ereg2.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, ereg3.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, ereg4.predict(X_test)),3))
ls1.append(round(mean_absolute_error(y_test, ereg5.predict(X_test)),3))

ls2.append(round(mean_absolute_error(y_train, lr.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, dt.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, gbr.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, rfr.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, ela.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, rg.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, las.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, xg.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, cb.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, ereg1.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, ereg2.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, ereg3.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, ereg4.predict(X_train)),3))
ls2.append(round(mean_absolute_error(y_train, ereg5.predict(X_train)),3))

ls1 = np.array(ls1)
ls1 = ls1.reshape(-1, 14)
result_test = pd.DataFrame(ls1, columns=cols1)

ls2 = np.array(ls2)
ls2 = ls2.reshape(-1, 14)
result_train = pd.DataFrame(ls2, columns=cols1)

C:\Users\admin\Anaconda3\lib\site-packages\ipykernel_launcher.py:15: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  from ipykernel import kernelapp as app
C:\Users\admin\Anaconda3\lib\site-packages\sklearn\utils\validation.py:63: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(*args, **kwargs)
C:\Users\admin\Anaconda3\lib\site-packages\ipykernel_launcher.py:17: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
C:\Users\admin\Anaconda3\lib\site-packages\sklearn\linear_model\_ridge.py:148: LinAlgWarning: Ill-conditioned matrix (rcond=4.44954e-19): result may not be accurate.
  overwrite_a=True).T


0:	learn: 363.8116887	total: 2.63ms	remaining: 261ms
1:	learn: 343.9088762	total: 4.81ms	remaining: 236ms
2:	learn: 330.8736390	total: 7.08ms	remaining: 229ms
3:	learn: 315.1429212	total: 9.28ms	remaining: 223ms
4:	learn: 301.3636623	total: 11.6ms	remaining: 220ms
5:	learn: 289.7586101	total: 13.9ms	remaining: 219ms
6:	learn: 277.8403404	total: 16.3ms	remaining: 216ms
7:	learn: 267.4940860	total: 18.7ms	remaining: 215ms
8:	learn: 254.1581110	total: 20.8ms	remaining: 211ms
9:	learn: 243.7915482	total: 23ms	remaining: 207ms
10:	learn: 235.9373266	total: 25.4ms	remaining: 205ms
11:	learn: 229.8138942	total: 28.4ms	remaining: 208ms
12:	learn: 224.1769732	total: 30.5ms	remaining: 204ms
13:	learn: 218.2796520	total: 32.4ms	remaining: 199ms
14:	learn: 211.9766735	total: 34.6ms	remaining: 196ms
15:	learn: 205.9144386	total: 36.6ms	remaining: 192ms
16:	learn: 199.3189269	total: 38.8ms	remaining: 189ms
17:	learn: 195.4989965	total: 40.7ms	remaining: 185ms
18:	learn: 191.2881807	total: 42.8ms	rem

C:\Users\admin\Anaconda3\lib\site-packages\sklearn\utils\validation.py:63: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(*args, **kwargs)
C:\Users\admin\Anaconda3\lib\site-packages\sklearn\utils\validation.py:63: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(*args, **kwargs)
C:\Users\admin\Anaconda3\lib\site-packages\sklearn\utils\validation.py:63: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(*args, **kwargs)


0:	learn: 363.8116887	total: 2.69ms	remaining: 266ms
1:	learn: 343.9088762	total: 5.48ms	remaining: 269ms
2:	learn: 330.8736390	total: 7.87ms	remaining: 254ms
3:	learn: 315.1429212	total: 10.2ms	remaining: 245ms
4:	learn: 301.3636623	total: 12.5ms	remaining: 238ms
5:	learn: 289.7586101	total: 14.9ms	remaining: 233ms
6:	learn: 277.8403404	total: 17.2ms	remaining: 229ms
7:	learn: 267.4940860	total: 19.7ms	remaining: 226ms
8:	learn: 254.1581110	total: 21.9ms	remaining: 222ms
9:	learn: 243.7915482	total: 24.5ms	remaining: 221ms
10:	learn: 235.9373266	total: 26.9ms	remaining: 218ms
11:	learn: 229.8138942	total: 29.1ms	remaining: 213ms
12:	learn: 224.1769732	total: 31.3ms	remaining: 210ms
13:	learn: 218.2796520	total: 34.2ms	remaining: 210ms
14:	learn: 211.9766735	total: 36.5ms	remaining: 207ms
15:	learn: 205.9144386	total: 38.7ms	remaining: 203ms
16:	learn: 199.3189269	total: 41ms	remaining: 200ms
17:	learn: 195.4989965	total: 43.7ms	remaining: 199ms
18:	learn: 191.2881807	total: 46.4ms	rem

C:\Users\admin\Anaconda3\lib\site-packages\sklearn\utils\validation.py:63: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(*args, **kwargs)
C:\Users\admin\Anaconda3\lib\site-packages\sklearn\utils\validation.py:63: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(*args, **kwargs)


In [65]:
result_test

Linear  ExtraTrees  Gradient  RandomForest  ElasticNet    Ridge   Lasso  \
0  106.978     115.899   123.997       117.671      101.58  104.659  99.874   

       XGB  Catboost  Voting1  Voting2  Voting3  Voting4  Voting5  
0  126.422   122.135   99.643    100.6  102.468   97.123  103.486

In [66]:
result_train

Linear  ExtraTrees  Gradient  RandomForest  ElasticNet    Ridge    Lasso  \
0  103.853         0.0    49.446        44.624     118.738  107.705  110.521   

     XGB  Catboost  Voting1  Voting2  Voting3  Voting4  Voting5  
0  0.247    69.112   75.023   87.665   95.557   74.984   89.641

### 변수중요도 확인

In [52]:
#Gradient
feature_gbr = pd.DataFrame({'Gradient Boost' : X.columns, '변수중요도':gbr.feature_importances_})
feature_gbr = feature_gbr.sort_values('변수중요도', ascending=False).reset_index(drop=True)
#ExtraTrees
feature_dt = pd.DataFrame({'ExtraTrees' : X.columns, '변수중요도':dt.feature_importances_})
feature_dt = feature_dt.sort_values('변수중요도', ascending=False).reset_index(drop=True)
#Catboost
feature_cat = pd.DataFrame({'Catboost' : X.columns, '변수중요도':cb.feature_importances_})
feature_cat = feature_cat.sort_values('변수중요도', ascending=False).reset_index(drop=True)
#Xgboost
feature_xg = pd.DataFrame({'XGB' : X.columns, '변수중요도':xg.feature_importances_})
feature_xg = feature_xg.sort_values('변수중요도', ascending=False).reset_index(drop=True)
#RandomForest
feature_rfr = pd.DataFrame({'RandomForest' : X.columns, '변수중요도':rfr.feature_importances_})
feature_rfr = feature_rfr.sort_values('변수중요도', ascending=False).reset_index(drop=True)
#변수중요도 데이터프레임
feature = pd.concat([feature_cat, feature_rfr, feature_xg, feature_dt, feature_gbr], axis=1)

In [53]:
feature

Catboost      변수중요도 RandomForest     변수중요도         XGB         변수중요도  \
0      단지내주차면수  27.974759      단지내주차면수  0.563962     단지내주차면수  2.690195e-01   
1          총면적  21.012296          총면적  0.222827       대전광역시  1.253529e-01   
2        규모별가격  12.812154        규모별가격  0.027713         총면적  1.103567e-01   
3   전용면적_25_비율   3.577833         총세대수  0.015322         A비율  6.710237e-02   
4         총세대수   2.683418   전용면적_25_비율  0.015105     70대(남자)  6.105294e-02   
5      ABEGH비율   2.585608          공가수  0.014930       아파트비율  5.304211e-02   
6      80대(남자)   1.899604   전용면적_35_비율  0.012750    JKLMNO비율  4.899129e-02   
7   전용면적_35_비율   1.804313   전용면적_45_비율  0.007967     50대(남자)  2.806111e-02   
8      70대(남자)   1.744006   전용면적_50_비율  0.007805        장기비율  2.275227e-02   
9          공가수   1.537104      40대(남자)  0.007094   10대미만(여자)  2.271154e-02   
10     40대(여자)   1.437015      80대(남자)  0.005471     40대(남자)  1.955921e-02   
11         A비율   1.303885        버스정류장  0.005327     20대(남자)  1.843438e-02   
12     50대(여자)   1.283817   전용면적_55_비율  0.004749       초장기비율  1.535145e-02   
13     20대(남자)   1.174396   전용면적_70_비율  0.004343  전용면적_50_비율  1.059298e-02   
14     ACEFI비율   1.025079          A비율  0.003950  전용면적_25_비율  1.046280e-02   
15   운전면허소지자비율   0.981280      20대(남자)  0.003837         연면적  9.461919e-03   
16  전용면적_55_비율   0.926011    운전면허소지자비율  0.003717  전용면적_30_비율  8.079645e-03   
17  전용면적_50_비율   0.922697    10대미만(여자)  0.003602     10대(남자)  7.300888e-03   
18    100대(여자)   0.918270      30대(남자)  0.003537        지하철역  7.246882e-03   
19       초장기비율   0.894540   전용면적_80_비율  0.003403     80대(남자)  7.118584e-03   
20  전용면적_80_비율   0.885237      50대(여자)  0.003290     자전거도로비율  6.057231e-03   
21        단기비율   0.691662      50대(남자)  0.003247  전용면적_70_비율  5.565030e-03   
22  전용면적_30_비율   0.572722        초장기비율  0.003088     30대(여자)  5.416316e-03   
23         연면적   0.566541      80대(여자)  0.002979  전용면적_35_비율  5.301669e-03   
24       아파트비율   0.561599      30대(여자)  0.002917  전용면적_80_비율  5.092173e-03   
25     90대(남자)   0.559144      70대(남자)  0.002793         공가수  4.887076e-03   
26    100대(남자)   0.542036   전용면적_75_비율  0.002753       규모별가격  4.621933e-03   
27        장기비율   0.462163      90대(남자)  0.002294  전용면적_75_비율  4.384041e-03   
28     20대(여자)   0.414831      90대(여자)  0.002158        총세대수  3.725204e-03   
29     90대(여자)   0.409087      ACEFI비율  0.002156     30대(남자)  3.425915e-03   
30        지하철역   0.385081     100대(여자)  0.002051    100대(여자)  3.414630e-03   
31      1인당자동차   0.323609      20대(여자)  0.002048  전용면적_45_비율  3.144358e-03   
32     자전거도로비율   0.317091     100대(남자)  0.001976     20대(여자)  3.082129e-03   
33   10대미만(여자)   0.314720   전용면적_30_비율  0.001910       버스정류장  2.032399e-03   
34  전용면적_15_비율   0.306577          연면적  0.001886     ACEFI비율  1.858313e-03   
35  전용면적_45_비율   0.301506     JKLMNO비율  0.001845      1인당자동차  1.681098e-03   
36     60대(여자)   0.294103      자전거도로비율  0.001820     40대(여자)  1.626054e-03   
37     30대(남자)   0.286838      40대(여자)  0.001712   운전면허소지자비율  1.550384e-03   
38       버스정류장   0.281711      ABEGH비율  0.001595   10대미만(남자)  1.306935e-03   
39  전용면적_75_비율   0.267854      10대(남자)  0.001423    100대(남자)  1.226751e-03   
40     50대(남자)   0.257607      70대(여자)  0.001227     10대(여자)  1.066215e-03   
41     30대(여자)   0.249106         상가비율  0.001198     50대(여자)  9.934425e-04   
42   10대미만(남자)   0.242228         충청남도  0.001194  전용면적_15_비율  8.934034e-04   
43     10대(여자)   0.230999    10대미만(남자)  0.001139     90대(여자)  8.826762e-04   
44  전용면적_70_비율   0.199961      10대(여자)  0.001090  전용면적_55_비율  8.721479e-04   
45         D비율   0.184350       평균분양가격  0.001033     90대(남자)  7.405771e-04   
46       서울특별시   0.176773        대전광역시  0.000953     80대(여자)  7.343029e-04   
47  전용면적_40_비율   0.167666   전용면적_20_비율  0.000888     60대(남자)  6.933055e-04   
48  전용면적_20_비율   0.145091          D비율  0.000880  전용면적_20_비율  5.223200e-04   
49     70대(여자)   0.137772         단기비율  0.000862  전용면적_40_비율  3.197357e-04   
50      평균분양가격   0.136805   

In [55]:
feature.to_csv('변수중요도.csv', index=False)

## 5. 모델 저장

In [67]:
with open('model_dash.pickle', 'wb') as fp:
    pickle.dump(ereg1, fp)
print('저장완료')

저장완료


## 6. Submission 만들기

### 1) 데이터 불러오기

In [68]:
df = pd.read_csv('data/test_ratio_0728.csv')

In [69]:
df.iloc[1,:]

단지코드                C1006
총세대수               1505.0
아파트비율            0.986711
상가비율             0.013289
공가수                  27.0
지하철역                  2.0
버스정류장                 5.0
단지내주차면수             428.0
전용면적_15_비율            0.0
전용면적_20_비율            0.0
전용면적_25_비율       0.784718
전용면적_30_비율       0.198007
전용면적_35_비율       0.008638
전용면적_40_비율            0.0
전용면적_45_비율            0.0
전용면적_50_비율       0.007309
전용면적_55_비율            0.0
전용면적_60_비율            0.0
전용면적_70_비율            0.0
전용면적_75_비율       0.000664
전용면적_80_비율       0.000664
초장기비율            0.986711
장기비율                  0.0
단기비율             0.013289
ABEGH비율               0.0
A비율                   0.0
ACEFI비율          0.986711
D비율              0.013289
JKLMNO비율              0.0
10대미만(여자)        0.028197
10대미만(남자)        0.029092
10대(여자)           0.04049
10대(남자)          0.042793
20대(여자)          0.060834
20대(남자)          0.064247
30대(여자)          0.068654
30대(남자)          0.066848
40대(여자)          0.074667
40대(남자)     

In [77]:
model.predict(df.iloc[1,:])

ValueError: could not convert string to float: 'C1006'

### 2) 모델 부르기

In [71]:
with open('model_ratio_0728.pickle', 'rb') as fp:
    model = pickle.load(fp)

### 3) 단지코드 제외

In [78]:
X = df.iloc[:, 1:]

In [85]:
X.shape

(147, 72)

In [84]:
X.iloc[1, :].values.reshape(-1,1)

array([[1.50500000e+03],
       [9.86710963e-01],
       [1.32890365e-02],
       [2.70000000e+01],
       [2.00000000e+00],
       [5.00000000e+00],
       [4.28000000e+02],
       [0.00000000e+00],
       [0.00000000e+00],
       [7.84717608e-01],
       [1.98006645e-01],
       [8.63787375e-03],
       [0.00000000e+00],
       [0.00000000e+00],
       [7.30897010e-03],
       [0.00000000e+00],
       [0.00000000e+00],
       [0.00000000e+00],
       [6.64451827e-04],
       [6.64451827e-04],
       [9.86710963e-01],
       [0.00000000e+00],
       [1.32890365e-02],
       [0.00000000e+00],
       [0.00000000e+00],
       [9.86710963e-01],
       [1.32890365e-02],
       [0.00000000e+00],
       [2.81970146e-02],
       [2.90916320e-02],
       [4.04897202e-02],
       [4.27925316e-02],
       [6.08339822e-02],
       [6.42467819e-02],
       [6.86536008e-02],
       [6.68477991e-02],
       [7.46674177e-02],
       [6.79246533e-02],
       [8.57507331e-02],
       [6.88192707e-02],


In [83]:
model.predict(X.iloc[1, :].values.reshape(-1,1))

ValueError: Feature shape mismatch, expected: 72, got 72

In [326]:
X.shape

(147, 72)

### 4) submission 생성

In [327]:
submission17 = pd.DataFrame({'code' : df.단지코드, 'predict' : model.predict(X)}) #비율만 넣은 것 test_2_semi

In [328]:
sub = pd.read_csv('sample_submission.csv')
sub = sub.merge(submission17, on='code', how='left')
sub = sub.drop('num', axis=1)
sub.columns = ['code', 'num']
sub.to_csv('submission/submission17.csv', index=False)

## Submission 정보 기입
- submission12 : xgboost, lasso, elasticnet, train_ratio.csv, test_ratio.csv, Only비율 106.7 
    - model_ratio.pickle
- submission13 : xgboost, lasso, elasticnet, 비율 + 여부 컬럼 108
    - model_ereg1_ratio_count.pickle
- submission14 : xgboost, lasso, elasticnet, submission12을 RobustScaler로 스케일링 121
    - model_ereg1_ratio_scaled.pickle
- submission15 : catboost, lasso, elasticnet, train_ratio.csv, test_ratio.csv, 모델 변경(catboost) 114
    - model_ratio_cat.pickle
- submission16 : xgboost, lasso, elasticnet, train_ratio.csv, test_ratio.csv, Only비율 106.7 (=submission12)
    - model_ratio.pickle
- submission17 : xgboost, lasso, elasticnet, train_ratio_0728, test_ratio_0728, 전용면적비율 추가, 자전거도로, 운전면허소지자현황 101.194
    - model_ratio_0728.pickle